tests/lib/types/types.test.ts

```diff
@@ -2121,6 +2126,25 @@ ruleTester.run("simple-valid-test", rule2, {
 	},
 });
 
+(): Linter.Config<{ foo?: Linter.RuleEntry<["always" | "never"]> }> => ({
+	rules: {
+		foo: ["error"],
+	},
+});
+
+// ensure Linter.Config is using RulesConfig
+declare module "@eslint/core" {
+	interface RulesConfig {
+		testExtensionRule?: RuleConfig<["always" | "never"]>;
+	}
+}
+
+(): Linter.Config => ({
+	rules: {
+		testExtensionRule: ["error"],
+	},
+});
+
```

Comment:
```suggestion

(): Linter.Config => ({
	rules: {
		testExtensionRule: ["error", "always"],
	},
});

(): Linter.Config => ({
	rules: {
		// @ts-expect-error // wrong value
		testExtensionRule: ["error", "foo"],
	},
});

(): Linter.Config => ({
	rules: {
		// @ts-expect-error // extra item
		testExtensionRule: ["error", "always", "never"],
	},
});
```

To ensure Linter.Config is using RulesConfig.